pkg/csi/service/driver.go

```diff
@@ -119,18 +120,33 @@ func (s *service) BeforeServe(
 	}
 
 	// Get the SP's operating mode.
-	s.mode = csictx.Getenv(ctx, gocsi.EnvVarMode)
-	if !strings.EqualFold(s.mode, "node") {
+	driver.mode = os.Getenv(csitypes.EnvVarMode)
+	if !strings.EqualFold(driver.mode, "node") {
 		// Controller service is needed
 		cfg, err = common.GetConfig(ctx)
 		if err != nil {
 			log.Errorf("failed to read config. Error: %+v", err)
 			return err
 		}
-		if err := s.cnscs.Init(cfg, Version); err != nil {
+		if err := driver.cnscs.Init(cfg, Version); err != nil {
 			log.Errorf("failed to init controller. Error: %+v", err)
 			return err
 		}
 	}
 	return nil
 }
+
+func (driver *vsphereCSIDriver) Run(ctx context.Context, endpoint string) {
+	log := logger.GetLogger(ctx)
+	controllerServer := driver.GetController()
+
+	// Invoke BeforeServe function to perform any local initialization routines.
+	if err := driver.BeforeServe(ctx, nil, nil); err != nil {
```

Comment:
Can we refactor this?  passing nil, nil args with just a context does not look good.

Reply:
I agree. But we also invoke this method when we use gocsi, and in that workflow it needs 3 params. See for reference: 
https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/d73639381c899762c715d489a7a6a732357af1e2/pkg/csi/provider/provider.go#L34

We can refactor it once we get rid of gocsi completely after testing.